sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -56,7 +53,7 @@ case class GpuUnaryPositive(child: Expression) extends GpuUnaryExpression
 
   override def sql: String = s"(+ ${child.sql})"
 
-  override def doColumnar(input: GpuColumnVector) : GpuColumnVector = input
+  override def doColumnar(input: GpuColumnVector) : ColumnVector = input.getBase.incRefCount()
```

Comment:
Sorry that was a bug.  This code never gets called in practice. `UnaryPositive` always becomes a noop and is filtered out of the plan.  We have it here just because.  The contract for `doColumnar` in this case is that the caller will close input, so you don't have to worry about it.